### ISSUE TITLE
Build failing while integrating AR Foundation(AR Core) to Flutter

### ISSUE THREAD

**Issue by GokulDas027** [snowballdigital/flutter-unity-view-widget#101]
FAILURE: Build failed with an exception.

What went wrong:
Could not determine the dependencies of task ':app:preDebugBuild'.
Could not resolve all task dependencies for configuration ':app:debugRuntimeClasspath'.
Could not find :arcore_client:.
Required by:
project :app > project :flutter_unity_widget > project :UnityExport
Could not find :unityandroidpermissions:.
Required by:
project :app > project :flutter_unity_widget > project :UnityExport
Could not find :UnityARCore:.
Required by:
project :app > project :flutter_unity_widget > project :UnityExport

Added modules(all 4 files)
Project Structure:
![Capture](https://user-images.githubusercontent.com/22936485/73595302-09ebee80-453d-11ea-9ddd-3826811152ea.PNG)

Android/Settings.gradle : 
![Capture1](https://user-images.githubusercontent.com/22936485/73595311-1c662800-453d-11ea-95c9-72514a745d81.PNG)

**Comment by subjectdenied** [comment]
this is a bit more complicated to be honest, if you never worked with android and gradle before, but basically you have to do this
- create a module of everything inside the unityexport/libs folder (the *.aar files), so you have all of them as subfolders including a build.gradle with each of them within the android directory
- add this to unityexport/build.gradle:
    implementation project(':unity-classes')
    implementation project(':ARPresto')
    implementation project(':UnityARCore')
    implementation project(':arcore_client')
    implementation project(':unityandroidpermissions')

make sure once it is working, you save this file, as currently it is overwritten everytime you do a new export from unity using the flutter menu, the export kinda does the same, but it wants to include the libs directly (using implementation name) which never worked for me

**Comment by GokulDas027** [comment]
Thanks!! It Worked!🤩

Firstly, It really complicated.. I couldn't even keep count of the number of build.gradle files..!!

* I already had some implementations in my UnityExport/build.gradle : 
>  implementation(name: 'arcore_client', ext:'aar')
>  implementation(name: 'unityandroidpermissions', ext:'aar')
>  implementation(name: 'UnityARCore', ext:'aar')
And so i thought, It was dealt with... and now it worked as I replaced it. Also `ARPresto` is somehow not needed now.. didn't have that project or module.

Thanks now that it worked👍

**Comment by GokulDas027** [comment]
Now, this is how the build.gradle file is..

![Annotation 2020-02-02 100431](https://user-images.githubusercontent.com/22936485/73603027-78639780-45a3-11ea-9958-a1011a1e706b.png)

**Comment by GokulDas027** [comment]
Successfully created the release build but the app is crashing on startup, checking for the reason..

**Comment by subjectdenied** [comment]
use --no-shrink when building the app, had the same issue

**Comment by Jairolaya12x** [comment]
@GokulDas027 You create the build.grandle in the unityExport folder? because i export and don't have this file 
Thanks

**Comment by scaydogdu** [comment]
anyone unfamiliar with gradle can check out: https://stackoverflow.com/questions/16682847/how-to-manually-include-external-aar-package-using-new-gradle-android-build-syst

**Comment by LucasTejero** [comment]
I already use this plugin to made an IOS Ar app,but now im trying to create the android version i follow all of the steps in the docs but i have this error
`E/Unity   (11649): Failed to load '/data/app/com.artplacer.artplacer-Cmg6Ih1K7jWKRmvSGLeqAw==/lib/x86/libmain.so', the application will terminate.
E/flutter (11649): [ERROR:flutter/shell/platform/android/platform_view_android_jni_impl.cc(43)] java.lang.UnsatisfiedLinkError: dlopen failed: library "/data/app/com.artplacer.artplacer-Cmg6Ih1K7jWKRmvSGLeqAw==/lib/x86/libmain.so" not found`

i dont know how to solve it,im new in the android world

**Comment by subjectdenied** [comment]
> I already use this plugin to made an IOS Ar app,but now im trying to create the android version i follow all of the steps in the docs but i have this error
> `E/Unity (11649): Failed to load '/data/app/com.artplacer.artplacer-Cmg6Ih1K7jWKRmvSGLeqAw==/lib/x86/libmain.so', the application will terminate. E/flutter (11649): [ERROR:flutter/shell/platform/android/platform_view_android_jni_impl.cc(43)] java.lang.UnsatisfiedLinkError: dlopen failed: library "/data/app/com.artplacer.artplacer-Cmg6Ih1K7jWKRmvSGLeqAw==/lib/x86/libmain.so" not found`
> 
> i dont know how to solve it,im new in the android world

were you ever able to fix this? have the same issue now with latest plugin (2.x), since i started to use gradle 5.6.4 (which has to be used to even be able to compile with arcore using unity2020.2b5+)

**Comment by rahulmool** [comment]
If any of you have implemented flutter unity widget with arfoundation then please send me the link of project. The project they include in documentation is really old. Please help me.

**Comment by Jatin-Pawar** [comment]
s

**Comment by Jatin-Pawar** [comment]
I also received the same message and was confused why this is so.
After some searching found out a simple yet working method to solve this.
Simply follow the following steps:
1) Open the "android" folder as a project in Android Studio.
2) Now, try to import .aar files by File > New > New Module > Import .JAR/AAR Package > Select android/UnityExport/libs/arcore-client.aar (This should import the files without any error.)
3) After step 2, you can check your UnityExport/settings.gradle file to find out that `include` ':arcore_client'` has been added at the top. (If not kindly add it.)
4) Repeat the above 2 steps for all the .arr/.jar files in the UnityExport/libs folder.
This steps would solve these below given errors
`
Could not resolve all task dependencies for configuration ':app:debugRuntimeClasspath'.
Could not find :arcore_client:.
Required by:
project :app > project :flutter_unity_widget > project :UnityExport
Could not find :unityandroidpermissions:.
Required by:
project :app > project :flutter_unity_widget > project :UnityExport
Could not find :UnityARCore:.
Required by:
project :app > project :flutter_unity_widget > project :UnityExport
`
